client/me/form-base/with-form-base.jsx

```diff
@@ -106,15 +64,28 @@ const withFormBase = ( WrappedComponent ) => {
 			toggleSetting: this.toggleSetting,
 			updateSetting: this.updateSetting,
 			submitForm: this.submitForm,
-			formState: this.state,
+			hasUnsavedUserSettings: this.props.hasUnsavedUserSettings,
 		} );
 
 		render() {
-			return <WrappedComponent { ...this.props } { ...this.getFormBaseProps() } />;
+			return (
+				<>
+					<QueryUserSettings />
+					<WrappedComponent { ...this.props } { ...this.getFormBaseProps() } />
+				</>
+			);
 		}
 	}
 
-	return connect( null, { errorNotice, successNotice } )( localize( EnhancedComponent ) );
+	return connect(
+		( state ) => ( {
+			_userSettings: getUserSettings( state ),
+			_unsavedUserSettings: getUnsavedUserSettings( state ),
+			hasUnsavedUserSettings: hasUnsavedUserSettings( state ),
```

Comment:
Note: I've noticed a pre-existing bug when making a change to the form and then reverting it. The form gets disabled, but when I try to navigate away from the page, I get the alert. Is this something we could fix while we're here? Not necessarily in this PR, of course.

Reply:
good catch! fixed it here a9a666e 👍🏻